constantine-go/constantine.go

```diff
@@ -0,0 +1,180 @@
+		err = errors.New(
+			C.GoString(C.ctt_eth_trusted_setup_status_to_string(status)),
+		)
+	}
+	return ctx, err
+}
+
+func (ctx EthKzgContext) Delete() {
+	C.ctt_eth_trusted_setup_delete(ctx.cCtx)
+}
+
+func (ctx EthKzgContext) BlobToKZGCommitment(blob EthBlob) (commitment EthKzgCommitment, err error) {
+	status := C.ctt_eth_kzg_blob_to_kzg_commitment(
+		ctx.cCtx,
+		(*C.ctt_eth_kzg_commitment)(unsafe.Pointer(&commitment)),
+		(*C.ctt_eth_kzg_blob)(unsafe.Pointer(&blob)),
+	)
+	if status != C.cttEthKzg_Success {
+		err = errors.New(
+			C.GoString(C.ctt_eth_kzg_status_to_string(status)),
+		)
+	}
+	return commitment, err
+}
+
+func (ctx EthKzgContext) ComputeKzgProof(blob EthBlob, z EthKzgChallenge) (proof EthKzgProof, y EthKzgEvalAtChallenge, err error) {
+	status := C.ctt_eth_kzg_compute_kzg_proof(
+		ctx.cCtx,
+		(*C.ctt_eth_kzg_proof)(unsafe.Pointer(&proof)),
+		(*C.ctt_eth_kzg_eval_at_challenge)(unsafe.Pointer(&y)),
+		(*C.ctt_eth_kzg_blob)(unsafe.Pointer(&blob)),
+		(*C.ctt_eth_kzg_challenge)(unsafe.Pointer(&z)),
+	)
+	if status != C.cttEthKzg_Success {
+		err = errors.New(
+			C.GoString(C.ctt_eth_kzg_status_to_string(status)),
+		)
+	}
+	return proof, y, err
+}
+
+func (ctx EthKzgContext) VerifyKzgProof(commitment EthKzgCommitment, z EthKzgChallenge, y EthKzgEvalAtChallenge, proof EthKzgProof) (bool, error) {
+	status := C.ctt_eth_kzg_verify_kzg_proof(
+		ctx.cCtx,
+		(*C.ctt_eth_kzg_commitment)(unsafe.Pointer(&commitment)),
+		(*C.ctt_eth_kzg_challenge)(unsafe.Pointer(&z)),
+		(*C.ctt_eth_kzg_eval_at_challenge)(unsafe.Pointer(&y)),
+		(*C.ctt_eth_kzg_proof)(unsafe.Pointer(&proof)),
+	)
+	if status != C.cttEthKzg_Success {
+		err := errors.New(
+			C.GoString(C.ctt_eth_kzg_status_to_string(status)),
+		)
+		return false, err
+	}
+	return true, nil
+}
+
+func (ctx EthKzgContext) ComputeBlobKzgProof(blob EthBlob, commitment EthKzgCommitment) (proof EthKzgProof, err error) {
+	status := C.ctt_eth_kzg_compute_blob_kzg_proof(
+		ctx.cCtx,
+		(*C.ctt_eth_kzg_proof)(unsafe.Pointer(&proof)),
+		(*C.ctt_eth_kzg_blob)(unsafe.Pointer(&blob)),
+		(*C.ctt_eth_kzg_commitment)(unsafe.Pointer(&commitment)),
+	)
+	if status != C.cttEthKzg_Success {
+		err = errors.New(
+			C.GoString(C.ctt_eth_kzg_status_to_string(status)),
+		)
+	}
+	return proof, err
+}
+
+func (ctx EthKzgContext) VerifyBlobKzgProof(blob EthBlob, commitment EthKzgCommitment, proof EthKzgProof) (bool, error) {
+	status := C.ctt_eth_kzg_verify_blob_kzg_proof(
+		ctx.cCtx,
+		(*C.ctt_eth_kzg_blob)(unsafe.Pointer(&blob)),
+		(*C.ctt_eth_kzg_commitment)(unsafe.Pointer(&commitment)),
+		(*C.ctt_eth_kzg_proof)(unsafe.Pointer(&proof)),
+	)
+	if status != C.cttEthKzg_Success {
+		err := errors.New(
+			C.GoString(C.ctt_eth_kzg_status_to_string(status)),
+		)
+		return false, err
+	}
+	return true, nil
+}
+
+func (ctx EthKzgContext) VerifyBlobKzgProofBatch(blobs []EthBlob, commitments []EthKzgCommitment, proofs []EthKzgProof, secureRandomeBytes [32]byte) (bool, error) {
```

Comment:
Also needs to be fixed here:
```suggestion
func (ctx EthKzgContext) VerifyBlobKzgProofBatch(blobs []EthBlob, commitments []EthKzgCommitment, proofs []EthKzgProof, secureRandomBytes [32]byte) (bool, error) {
```